server/histogram-aggregator.go

```diff
@@ -22,16 +23,23 @@ type histAggregator struct {
 	// usec.
 	// Initialized lazily in AddValue().
 	h *hdrhistogram.Histogram
+	m sync.Mutex
 }
 
 func (a *histAggregator) AddValue(value float64, ts int64) {
+	a.m.Lock()
+	defer a.m.Unlock()
```

Comment:
It's good that you noticed that AddValue and GetAggregators are called from different coroutines. Let's add a comment on the aggregatorUnit interface that explicitly calls out that these must be thread-safe w.r.t. each other.